src/v/redpanda/admin/server.cc

```diff
@@ -159,6 +160,8 @@
 
```

Comment:
Did you measure compile time changes with this? 

Can we maybe remove some includes here now as well to further reduce compile times and cut the dependency chain for some headers.

Reply:
Not yet, hopefully we can see a difference as we continue to make progress on breaking this up. We can probably do a pass and remove headers here once we've made more progress.

Reply:
If we split things up but don't remove includes are we not making things worse compared to where we are now?

Reply:
How is it worse? We can parallelize and have better caching with the breakup. I was able to remove a few headers, but not that much yet.

Reply:
> How is it worse?

Well it depends on why compiling application.cc, debug.cc, server.cc etc is so slow. 

Is it because of the actual code in those files or because of the plethora of includes and everything that comes with it. 

If it's the later then with every additional file you add and hence include the same files again then it would just add overhead if not removing them from where the includes are no longer needed. In the end it might not matter because the total compile time is limited by those mega files anyway.

Reply:
I suspect it is related to headers. I think there are some poison pill headers somewhere that really cause us issues. application.cc and server.cc are particularly special in that they effectively include everything.